pallets/claims/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+use codec::Encode;
+use frame_support::{
+	decl_error, decl_event, decl_module, decl_storage,
+	dispatch::DispatchResult,
+	ensure,
+	traits::Get,
+	weights::{DispatchClass, Pays},
+};
+use frame_system::ensure_signed;
+use hex::FromHex;
+use orml_traits::{MultiCurrency, MultiCurrencyExtended};
+use orml_utilities::with_transaction_result;
+use primitives::{Amount, AssetId, Balance, CORE_ASSET_ID};
+use sp_io::{crypto::secp256k1_ecdsa_recover, hashing::keccak_256};
+use sp_runtime::traits::Zero;
+use sp_std::prelude::*;
+use sp_std::vec::Vec;
+
+pub use traits::*;
+
+mod claims_data;
+mod traits;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+mod benchmarking;
+
+use weights::WeightInfo;
+pub mod weights;
+
+pub trait Config: frame_system::Config {
+	type Event: From<Event<Self>> + Into<<Self as frame_system::Config>::Event>;
+	type Currency: MultiCurrencyExtended<Self::AccountId, CurrencyId = AssetId, Balance = Balance, Amount = Amount>;
+	type Prefix: Get<&'static [u8]>;
+	type WeightInfo: WeightInfo;
+}
+
+type BalanceOf<T> = <<T as Config>::Currency as MultiCurrency<<T as frame_system::Config>::AccountId>>::Balance;
+
+decl_storage! {
+	trait Store for Module<T: Config> as Claims {
+		HDXClaims get(fn hdxclaims): map hasher(blake2_128_concat) EthereumAddress => BalanceOf<T>;
```

Comment:
Pls rename this to ClaimData or something more indicative of what it is.

Reply:
IMHO, the "Data" does not provide more clarity but I can refactor the storage name and remove HDX from it. "Claims" as the name of storage is used for multiple projects including Polkadot, so I would stick with that.